### PR TITLE
VK: fetch metadata and state from objects instead of storage

### LINKED ISSUE #1162: Vk.com: suddenly stops scrobbling
Hello! Today I saw thats scrobbling stop working on Vk.com. I'm using Google Chrome 54.0.2840.99 m.

### PR DESCRIPTION
**Pull request by f1u77y**
These objects(`audio` and `ap._impl._currentAudioEl`) seem to be more stable
then 'audio_vXX_whatever' localStorage items

Fixes #1162

### CODE REVIEW

File: connectors/v2/vk.js [hunk 1]
```diff
@@ -2,62 +2,47 @@
...
+		if (typeof event.data.type !== 'string') {
 			return;
 		}
-		if (eventType.startsWith('vk:player')) {
-			if (eventType.endsWith('start')) {
+		if (event.data.type.startsWith('vk:player')) {
+			if (event.data.type.endsWith('start')) {
 				isPlaying = true;
-			} else if (eventType.endsWith('stop') || eventType.endsWith('pause')) {
+			} else if (event.data.type.endsWith('stop') || event.type.endsWith('pause')) {
```
Comment (alexesprit) [inline connectors/v2/vk.js]: It should be `event.data.type` in the second condition.

**Comment by f1u77y** [comment]
Whoops, tabs

**Comment by alexesprit** [comment]
`lintspaces` is configured to disallow spaces as indentation.

**Comment by f1u77y** [comment]
@alexesprit I'm talking about indenting with tabs and aligning with spaces

**Comment by alexesprit** [comment]
@f1u77y I understood you, but lintspaces assumes that spaces is an indentation, not an alignment.

P.S. Mixed tabs and spaces is not that thing should be used. That's why spaces are better for indentation.

**Comment by f1u77y** [comment]
> Mixed tabs and spaces is not that thing should be used.

But that's the only way to align code with tabs in tab-size-agnostic way.

Sigh...

**Comment by alexesprit** [comment]
> But that's the only way to align code with tabs in tab-size-agnostic way.

No, because tab width is configurable.

**Comment by f1u77y** [comment]
Anyway, I've updated the PR to don't use spaces for alignment

(and I don't think that PR discussion is suitable place for yet another tabs-vs-spaces arguement)

**Pull request merged**